pipeline_components/aiplatform/aiplatform/model_batch_predict/component.py

```diff
@@ -0,0 +1,130 @@
+        job_display_name: Name of the batch prediction job.
+        project_location (str): location of the Google Cloud project. Defaults to None.
+        project_id (str): project id of the Google Cloud project. Defaults to None.
+        source_uri (str): bq:// URI or a list of gcs:// URIs to read input instances.
+        destination_uri (str): bq:// or gcs:// URI to store output predictions.
+        source_format (str): E.g. "bigquery", "jsonl", "csv". Reference:
+            https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1beta1.types.BatchPredictionJob.InputConfig
+        destination_format (str): E.g. "bigquery", "jsonl", "csv". Reference:
+            https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1beta1.types.BatchPredictionJob.OutputConfig
+        machine_type (str): Machine type.
+        starting_replica_count (int): Starting replica count.
+        max_replica_count (int): Max replicat count.
+        alert_email_addresses (List[str]): Email addresses to send alerts to (optional).
+    Returns:
+        NamedTuple: gcp_resources for Vertex AI UI integration.
+    """
+
+    import json
+    import logging
+    from google.protobuf.json_format import ParseDict, MessageToJson
+    from google.cloud.aiplatform_v1beta1.services.job_service import JobServiceClient
+    from google.cloud.aiplatform_v1beta1.types import BatchPredictionJob
+    from google_cloud_pipeline_components.proto.gcp_resources_pb2 import GcpResources
+
+    api_endpoint = f"{project_location}-aiplatform.googleapis.com"
+    if alert_email_addresses is None:
+        alert_email_addresses = []
+
+    TRAINING_DATASET_INFO = "training_dataset.json"
+    logging.info(f"Read {model.path + '/' + TRAINING_DATASET_INFO}")
+    with open(model.path + "/" + TRAINING_DATASET_INFO, "r") as fp:
+        training_dataset = json.load(fp)
+
+    input_config = {"instancesFormat": source_format}
+    output_config = {"predictionsFormat": destination_format}
+    if source_format == "bigquery" and destination_format == "bigquery":
+        input_config["bigquerySource"] = {"inputUri": source_uri}
+        output_config["bigqueryDestination"] = {"outputUri": destination_uri}
+    else:
+        # TODO hotfix for wrong inputs from TF prediction pipeline
+        input_config["gcsSource"] = {"uris": json.loads(source_uri)}
+        output_config["gcsDestination"] = {"outputUriPrefix": destination_uri}
+
+    message = {
+        "displayName": job_display_name,
+        "model": model.metadata["resourceName"],
+        "inputConfig": input_config,
+        "outputConfig": output_config,
+        "dedicatedResources": {
+            "machineSpec": {"machineType": machine_type},
+            "startingReplicaCount": starting_replica_count,
+            "maxReplicaCount": max_replica_count,
+        },
+        "modelMonitoringConfig": {
```

Comment:
maybe we just make the whole of the `modelMonitoringConfig` block a dictionary input to the component?

Reply:
Added three new inputs:
- monitoring_training_dataset: dict = None,
- monitoring_alert_email_addresses: List[str] = None,
- monitoring_skew_config: dict = None,

The first one is passed by `lookup_component` (which won't be used in training in near future due to the CC changes) and the latter two can be set by users in the pipeline.

Reply:
Looks good - please can you update the docstring to reflect this